lib/jxl/render_pipeline/xyb_stage.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
```

Comment:
It would be useful to add some comments here or in the .cc file about what this stage is doing, and maybe what invariants/expectations/assumptions there are about the input/output.

It's also not clear to me if these implementations of stages will be shared between the simple and complicated pipeline or not. I would assume the whole point is that they can be shared (and the main difference will be that the simple pipeline calls them on full frame buffers in a simple chain while the complicated one does things per group and has to deal with borders and dependencies and keeping track of when to allocate what), but I might be wrong and it would be good to clarify this somewhere in a comment.

Reply:
Added a comment about what this does.

The whole point about the pipeline is that there should be no assumptions/preconditions/expectations needed for all its stages, so I don't think we need more comments about that.

As for sharing implementations: by default I'd assume that if two things are called a different implementation of an interface, they expose the same behaviour as far as that interface is concerned, so I don't see the need to clarify that in a comment.